src/NumericalAlgorithms/Spectral/SwshCollocation.hpp

```diff
@@ -0,0 +1,256 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstdlib>
+#include <memory>
+#include <sharp_cxx.h>
+#include <type_traits>
+#include <utility>
+
+#include "NumericalAlgorithms/Spectral/ComplexDataView.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+// In the static caching mechanism, we permit an l_max up to this macro
+// value. Higher l_max values may still be created manually using the
+// `Collocation` constructor. If l_max's are used several times at higher value,
+// consider increasing this value, but only after memory costs have been
+// evaluated.
+#define MAXIMUM_L_MAX 200
```

Comment:
I recommend "namespacing" this macro by naming it `SWSH_COLLOCATION_MAXIMUM_L_MAX`

Reply:
Quick question: Is there a reason this is a macro and not a constexpr variable?

Reply:
I don't have any particular reason for preferring a macro. I'll change it to a constexpr in the next fixup.